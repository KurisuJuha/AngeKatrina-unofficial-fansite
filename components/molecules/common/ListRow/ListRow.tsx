import React, { useState } from "react";
import styled from "styled-components";
import H2 from "../../../atomics/common/H2";

interface Props {
  mainText: string;
  IconSvg?: React.StatelessComponent<React.SVGAttributes<SVGElement>>;
  HoveredSvg?: React.StatelessComponent<React.SVGAttributes<SVGElement>>;
  bgColor: string;
  textColor: string;
  cursor?: "default" | "pointer";
  onClickFC?: () => void;
}

const Wrapper = styled.div<{
  bgColor: Props["bgColor"];
  textColor: Props["textColor"];
  cursor: Props["cursor"];
}>`
  width: 100%;
  height: 72px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  background-color: ${({ bgColor }) => bgColor};
  color: ${({ textColor }) => textColor};
  cursor: ${({ cursor }) => cursor};
  :hover {
    background-color: ${({ textColor }) => textColor};
    color: ${({ bgColor }) => bgColor};
  }
`;

const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 28px;
  width: 32px;
  margin-left: 24px;
  margin-right: 12px;
`;

const ListRow: React.FC<Props> = ({
  mainText,
  IconSvg,
  HoveredSvg,
  bgColor,
  textColor,
  cursor = "default",
  onClickFC,
}: Props) => {
  const [isHovering, changeIsHovering] = useState(false);
  return (
    <Wrapper
      bgColor={bgColor}
      textColor={textColor}
      onMouseEnter={() => changeIsHovering(true)}
      onMouseLeave={() => changeIsHovering(false)}
      cursor={cursor}
      onClick={onClickFC}
    >
      <IconWrapper>
        {HoveredSvg !== undefined && IconSvg !== undefined ? (
          isHovering ? (
            <HoveredSvg />
          ) : (
            <IconSvg />
          )
        ) : null}
      </IconWrapper>
      <H2 color={isHovering ? bgColor : textColor} text={mainText}></H2>
    </Wrapper>
  );
};

export default ListRow;
