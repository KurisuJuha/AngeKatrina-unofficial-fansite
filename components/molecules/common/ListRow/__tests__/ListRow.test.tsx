import React from "react";
import ListRow from "../ListRow";
import renderer from "react-test-renderer";

describe("snapshot test", () => {
  test("pass minimum props", () => {
    const component = renderer.create(
      <ListRow mainText="sample" bgColor={"#FFFFFF"} textColor={"#000000"} />
    );
    const tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  });

  test("pass cursor: pointer prop", () => {
    const component = renderer.create(
      <ListRow
        cursor="pointer"
        mainText="sample"
        bgColor={"#FFFFFF"}
        textColor={"#000000"}
      />
    );
    const tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  });

  test("pass cursor: default prop", () => {
    const component = renderer.create(
      <ListRow
        cursor="default"
        mainText="sample"
        bgColor={"#FFFFFF"}
        textColor={"#000000"}
      />
    );
    const tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  });
});
