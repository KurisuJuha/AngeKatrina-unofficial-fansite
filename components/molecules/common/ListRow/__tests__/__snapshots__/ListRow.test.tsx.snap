// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test pass cursor: default prop 1`] = `
<div
  className="ListRow__Wrapper-sc-1mu2w86-0 dOEazA"
  cursor="default"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="ListRow__IconWrapper-sc-1mu2w86-1 cgdwMp"
  />
  <h2
    className="H2__Text-cfm3jm-0 kMZDyH"
    color="#000000"
  >
    sample
  </h2>
</div>
`;

exports[`snapshot test pass cursor: pointer prop 1`] = `
<div
  className="ListRow__Wrapper-sc-1mu2w86-0 byiPZs"
  cursor="pointer"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="ListRow__IconWrapper-sc-1mu2w86-1 cgdwMp"
  />
  <h2
    className="H2__Text-cfm3jm-0 kMZDyH"
    color="#000000"
  >
    sample
  </h2>
</div>
`;

exports[`snapshot test pass minimum props 1`] = `
<div
  className="ListRow__Wrapper-sc-1mu2w86-0 dOEazA"
  cursor="default"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="ListRow__IconWrapper-sc-1mu2w86-1 cgdwMp"
  />
  <h2
    className="H2__Text-cfm3jm-0 kMZDyH"
    color="#000000"
  >
    sample
  </h2>
</div>
`;
